luthier/src/test/groovy/com/yahoo/bard/webservice/config/luthier/ResourceNodeSupplierTest.groovy

```diff
@@ -0,0 +1,92 @@
+        when:
+            ObjectNode defaultNode = defaultResourceNodeSupplier.get()
+            ObjectNode dimensionsNode = testResourceNodeSupplier.get()
+        then:
+            defaultNode == null
+            dimensionsNode != null
+    }
+
+    def "a LuthierFactoryException is thrown when .json is not formatted correctly"() {
+        given:
+            ResourceNodeSupplier nonExistentResourceNodeSupplier = new ResourceNodeSupplier("NON_EXISTENT.json")
+        when:
+            ObjectNode nonExistentNode = nonExistentResourceNodeSupplier.get()
+        then:
+            thrown(LuthierFactoryException)
+    }
+
+    def "All contents of a test dimension is correct"() {
+        when:
+            ObjectNode node = testResourceNodeSupplier.get().get("testDimension")
+            String apiName = node.get("apiName").textValue()
+            String longName = node.get("longName").textValue()
+            ArrayNode fields = node.get("fields")
+            String category = node.get("category").textValue()
+            String description = node.get("description").textValue()
+            String searchProvider = node.get("searchProvider").textValue()
+            String keyValueStore = node.get("keyValueStore").textValue()
+        then:
+            apiName == "testDimension"
+            longName == "a longName for testing"
+            fields.size() == 4
+            fields.get(0).get("name").textValue() == "TEST_PK"
+            fields.get(0).get("tags").get(0).textValue() == "primaryKey"
+            fields.get(1).get("name").textValue() == "TEST_FIELD_1"
+            fields.get(2).get("name").textValue() == "TEST_FIELD_2"
+            fields.get(3).get("name").textValue() == "TEST_FIELD_3"
+            category == "a category for testing"
+            description == "a description for testing"
+            searchProvider == "com.yahoo.bard.webservice.data.dimension.impl.NoOpSearchProvider"
+            keyValueStore == "com.yahoo.bard.webservice.data.dimension.MapStore"
+    }
+
+    def "All dimensions contain necessary keys (apiName, type, field, etc.), regardless their values"() {
+        when:
+            ObjectNode dimensionsNode = testResourceNodeSupplier.get()
+        then:
+            dimensionsNode.every {
+                it.has("type")
+                it.has("keyValueStore")
+                it.has("longName")
+                it.has("apiName")
+                it.has("searchProvider")
+                it.has("description")
+                it.has("fields")
+                it.get("fields").every {
+                    it.has("name")
+                }
+                // guarantees that at least one
+                // {
+                //    "tags": [
+                //       "primaryKey"
+                //    ],
+                //    "name": ...
+                // }
+                // exists in the fields value
+                it.get("fields").any {
+                    it.has("tags")
```

Comment:
This doesn't seem to be doing anything. Also, rather than using an if-statement, I think it'd
be clearer if you used `filter`:
```groovy
it.get("fields")
.filter {it.has("tags")
.any {it.get("tags").any {it.textValue() == "primaryKey"
```
`any` returns false I believe if you pass it an empty collection, thought that should be verified.

Reply:
The previous idea was that at least one of the field need to satisfy both (has "tags") and (content of "tags" contains primaryKey. Without the first one an empty "fields" will trivially pass.)

I agree to switch to the filter, and sense the design is changed (now any field has a "tags" list, albeit might be empty), the new logic will be different by a bit. It will be live shortly.

Reply:
I will actually not check primaryKey at all in this new commit, since that is not a syntax level check the ResourceNodeSupplier should test for. 

I will implement it right below the layer it is immediately dependent on.